### PR TITLE
validates title presence and uniqueness

### PR DESCRIPTION
**Pull request by fabersky**

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def has_friendly_id_slug?
   has_many :subscribers, through: :subscriptions, class_name: "User", source: :user
   has_many :endorsements, class_name: "ArticleEndorsement", counter_cache: true, dependent: :destroy
   has_many :endorsers, through: :endorsements, class_name: "User", source: :user
+  validates :title, presence: true, uniqueness: true, allow_nil: true
```
Comment (markets) [inline app/models/article.rb]: @SkyWDev it's a little weird validate `presence` and `allow_nil` on the same attribute.

File: app/views/articles/new.html.haml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-- if !@article.valid?
+- if !@article.valid? && !@article.blank?
```
Comment (markets) [inline app/views/articles/new.html.haml]: Is this change really necessary?

https://github.com/orientation/orientation/blob/master/app/controllers/articles_controller.rb#L23-L25

File: app/models/article.rb (hunk 1 above)
Comment (SkyWDev) [inline app/models/article.rb]: @markets it wasn't working without the `allow_nil: true`. When creating a new article, the warning "There's already an article with this title." was appearing even if the title was empty. Looking for a better solution!

**Comment by olivierlacan** [comment]
Going to close this since presence validation was added. We need to handle uniqueness at some point on the server-side and client-side (while people are typing) so if you want to tackle this feel free to update this PR and re-open it.
